mxnet/mxnet-engine/src/main/java/ai/djl/mxnet/jna/JnaUtils.java

```diff
@@ -1742,20 +1746,7 @@ public static synchronized CachedOp createCachedOp(
         String[] keys = {"data_indices", "param_indices", "static_alloc", "static_shape"};
         String[] values = {dataIndices.values().toString(), paramIndices.toString(), "1", "1"};
 
-        // thread-safe CachedOp doesn't support training
-        if (training) {
```

Comment:
This was originally for debug purpose. If we don't have the check, we might throw ugly engine exception. though if it is for performance concern, I agree to push the check to the engine itself